src/tvheadend/HTSPDemuxer.cpp

```diff
@@ -384,6 +386,39 @@ bool HTSPDemuxer::ProcessMessage ( const char *method, htsmsg_t *m )
   return true;
 }
 
+void HTSPDemuxer::ProcessRDS(uint32_t idx, const void* bin, size_t binlen)
+{
+  const uint8_t* data = static_cast<const uint8_t*>(bin);
+  const size_t offset = binlen - 1;
+
+  static const uint8_t RDS_IDENTIFIER = 0xFD;
+
+  if (data[offset] == RDS_IDENTIFIER)
+  {
+    // RDS data present, obtain length.
+    uint8_t rdslen = data[offset - 1];
+    if (rdslen > 0)
+    {
+      DemuxPacket* pkt = PVR->AllocateDemuxPacket(rdslen);
+      if (!pkt)
+        return;
+
+      uint8_t* rdsdata = new uint8_t[rdslen];
+
+      // Reassemble UECP block. mpeg stream contains data in reverse order!
+      for (int i = offset - 2, j = 0; i > 3 && i > offset - 2 - rdslen; i--, j++)
+        rdsdata[j] = data[i];
+
+      memcpy(pkt->pData, rdsdata, rdslen);
```

Comment:
Is rdslen always equal to the actual length of rdsdata? My brain melts when I look at the loop above.

Reply:
:-) yes.

Reply:
Hehe good